server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/restore/TransportRestoreSnapshotAction.java

```diff
@@ -77,18 +79,29 @@ protected ClusterBlockException checkBlock(RestoreSnapshotRequest request, Clust
     protected void masterOperation(final RestoreSnapshotRequest request,
                                    final ClusterState state,
                                    final ActionListener<RestoreSnapshotResponse> listener) {
-        RestoreService.RestoreRequest restoreRequest = new RestoreService.RestoreRequest(request.repository(), request.snapshot(),
-                request.indices(), request.templates(), request.indicesOptions(),
-                request.tableRenamePattern(), request.tableRenameReplacement(),
-                request.schemaRenamePattern(), request.schemaRenameReplacement(),
-                request.settings(), request.masterNodeTimeout(), request.partial(), request.includeAliases(),
-                request.indexSettings(), request.ignoreIndexSettings(), "restore_snapshot[" + request.snapshot() + "]",
-                request.includeIndices(),
-                request.includeCustomMetadata(),
-                request.customMetadataTypes(),
-                request.includeGlobalSettings(),
-                request.globalSettings());
-
+        RestoreService.RestoreRequest restoreRequest = new RestoreService.RestoreRequest(
+            request.repository(),
+            request.snapshot(),
+            Strings.EMPTY_ARRAY,
+            Strings.EMPTY_ARRAY,
+            request.indicesOptions(),
+            request.tableRenamePattern(),
+            request.tableRenameReplacement(),
+            request.schemaRenamePattern(),
+            request.schemaRenameReplacement(),
+            request.settings(),
+            request.masterNodeTimeout(),
+            false,
+            request.includeAliases(),
+            Settings.EMPTY,
+            Strings.EMPTY_ARRAY,
+            "restore_snapshot[" + request.snapshot() + "]",
+            request.includeIndices(),
+            request.includeCustomMetadata(),
+            request.customMetadataTypes(),
+            request.includeGlobalSettings(),
+            request.globalSettings()
+        );
```

Comment:
I think request.indices(), request.templates(), being an empty array was due to removal of BWC? However, we still need it for logical replication? Related: https://github.com/crate/crate/pull/17468/files#r1959909252

Reply:
Nevermind, it's hardcoded values from the logical replication usage